src/XIVLauncher.Common/Support/DalamudAndPluginConfigImportExport.cs

```diff
@@ -0,0 +1,121 @@
+using System;
+using System.Collections.Generic;
+using System.IO.Compression;
+using System.IO;
+using System.Threading.Tasks;
+
+namespace XIVLauncher.Common.Support
+{
+    public class DalamudAndPluginConfigImportExport
+    {
+        public static async Task<string> ExportConfig(string storagePath)
+        {
+            // grab any platform-specific details
+            // there's none to grab currently :)
+
+            // start making our zip
+            var outFile = new FileInfo(Path.Combine(storagePath, $"XIVLauncher-{DateTime.Now:yyyyMMddhhmmss}.xlconf"));
+            using var archive = ZipFile.Open(outFile.FullName, ZipArchiveMode.Create);
+
+            // add all the stock XIVLauncher based paths
+            var accountsListFile = Path.Combine(storagePath, "accountsList.json");
+            var dalamudConfigFile = Path.Combine(storagePath, "dalamudConfig.json");
+            var dalamudVfsFile = Path.Combine(storagePath, "dalamudVfs.db");
+            var pluginConfigsFolder = Path.Combine(storagePath, "pluginConfigs");
+
+            AddIfExist(accountsListFile, archive);
+            AddIfExist(dalamudConfigFile, archive);
+            AddIfExist(dalamudVfsFile, archive);
+            AddIfExist(pluginConfigsFolder, archive);
+
+            // add some known special exceptions. It might be better to not build these expectations though
+            var backupsFolder = Path.Combine(storagePath, "backups"); // Otter plugins
+            var playerTrackBackupsFolder = Path.Combine(storagePath, "playerTrackBackups"); // PlayerTrack
+
+            AddIfExist(backupsFolder, archive);
+            AddIfExist(playerTrackBackupsFolder, archive);
+
+            // return the folder containing our exported settings
+            return outFile.FullName;
+        }
+
+        public static void ImportConfig(string zipFilePath, string storagePath)
+        {
+            // grab any platform-specific details
+            // there's none to grab currently :)
+
+            // TODO: Decide if we're going to alert on overwriting config.
+            // Right now, Franz decided against it. The user has to intentionally try to use this feature
+            // Also, .Net Framework is dumb and will explode if we use ZipArchive.ExtractToDirectory()
+            // and there are any file conflicts. .Net Core doesn't have this issue though and provides
+            // an override we could have used to do it anyways. Alternatively, we could just delete
+            // all of the files/folders we'd be restoring to first, but that also feels bad.
+
+
+
+            var inFile = new FileInfo(zipFilePath);
+            using var archive = ZipFile.Open(inFile.FullName, ZipArchiveMode.Read);
+
+            // If we weren't on .Net Framework, we could use this...
+            // ZipFileExtensions.ExtractToDirectory(archive, storagePath, true);
```

Comment:
It's probably better to go through the entries directly. Then you can also choose to delete the file if it exists - which is probably what people expect when importing a backup.

Reply:
My initial vision was to just overwrite from the backup, but ideally yes, a "do you want to replace this file, yes, yes to all, no, cancel" style prompt would be better. My worry here is some plugins generate TONS of files in their config directories and most users are not going to want to click yes to that every time. (ex: CharacterSync making backups of all game config for multiple characters for multiple days/launches)

This was more of a companion to the official launcher's FEA file solution, but for our config. (Although nothing prevents this from later handling game config too if we wanted to do that)